src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,261 @@
+        int expectedSizeOfStorage = INITIAL_SIZE_OF_STORAGE + 1;
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(expectedSizeOfStorage, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_alreadyExistedUser_notOk() {
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(INITIAL_USER_IN_STORAGE);
+        });
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_nullUser_notOkay() {
+        User nullUser = null;
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(nullUser);
+        });
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(INITIAL_SIZE_OF_STORAGE, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_userWithValidLogin_ok() {
+        User expected = new User(1L, VALID_LOGIN_EXAMPLE, VALID_PASSWORD_EXAMPLE, MIN_VALID_AGE);
+        User actual = null;
+        try {
+            actual = registrationService.register(expected);
+        } catch (RegistrationException e) {
+            fail(e.getMessage());
+        }
+        assertEquals(expected, actual);
+        int expectedSizeOfStorage = INITIAL_SIZE_OF_STORAGE + 1;
+        int actualSizeOfStorage = Storage.people.size();
+        assertEquals(expectedSizeOfStorage, actualSizeOfStorage);
+    }
+
+    @Test
+    void register_userWithInvalidLogin_notOk() {
+        User fiveCharacterLoginUser = new User(1L, INVALID_LOGIN_EXAMPLE1,
+                VALID_PASSWORD_EXAMPLE,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(fiveCharacterLoginUser);
+        });
+
+        User fourCharacterLoginUser = new User(1L, INVALID_LOGIN_EXAMPLE2,
+                VALID_PASSWORD_EXAMPLE,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(fourCharacterLoginUser);
+        });
+
+        User oneCharacterLoginUser = new User(1L, INVALID_LOGIN_EXAMPLE3,
+                VALID_PASSWORD_EXAMPLE,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(oneCharacterLoginUser);
+        });
+
+        User emptyCharacterLoginUser = new User(1L, INVALID_LOGIN_EXAMPLE4,
+                VALID_PASSWORD_EXAMPLE,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(emptyCharacterLoginUser);
+        });
+
+        User sixWhiteSpacesLoginUser = new User(1L, INVALID_LOGIN_EXAMPLE5,
+                VALID_PASSWORD_EXAMPLE,
+                MIN_VALID_AGE);
+        assertThrows(RegistrationException.class, () -> {
+            registrationService.register(sixWhiteSpacesLoginUser);
+        });
```

Comment:
I think that you don't have to check every possible length of login. 

>  Few not-ok checks you can have in one not-ok test method. Same for ok-test. Testing only ‘abc’ (3-length - not-ok) and ‘abcdefgh’ (8-length - ok) is not enough because we do not check how the program behave with edge cases.

You can check if it is less than 6 symblos, if it equals and is more than 6 and if it is null.